pypowsybl/perunitnetwork.py

```diff
@@ -0,0 +1,636 @@
+        loads = self.get_elements(_pypowsybl.ElementType.LOAD)
+        loads['p0'] /= self.sn
+        loads['q0'] /= self.sn
+        loads['p'] /= self.sn
+        loads['q'] /= self.sn
+        join = _pd.merge(loads, self.get_voltage_levels()['nominal_v'],
+                         left_on='voltage_level_id', right_index=True)
+        join['i'] /= self.sn * 10 ** 3 / (self.sqrt3 * join['nominal_v'])
+        return join.drop('nominal_v', axis=1)
+
+    def get_lines(self) -> _pd.DataFrame:
+        """ Get lines as a ``Pandas`` data frame.
+
+        Returns:
+            a lines data frame
+        """
+        lines = self.get_elements(_pypowsybl.ElementType.LINE)
+        lines['p1'] /= self.sn
+        lines['p2'] /= self.sn
+        lines['q1'] /= self.sn
+        lines['q2'] /= self.sn
+        lines = _pd.merge(lines, self.get_voltage_levels()['nominal_v'],
+                          left_on='voltage_level1_id', right_index=True)
+        lines['i1'] /= self.sn * 10 ** 3 / (self.sqrt3 * lines['nominal_v'])
+        lines['i2'] /= self.sn * 10 ** 3 / (self.sqrt3 * lines['nominal_v'])
+        lines['r'] /= lines['nominal_v'] ** 2 / self.sn
+        lines['x'] /= lines['nominal_v'] ** 2 / self.sn
+        lines['g1'] /= self.sn / lines['nominal_v'] ** 2
+        lines['g2'] /= self.sn / lines['nominal_v'] ** 2
+        lines['b1'] /= self.sn / lines['nominal_v'] ** 2
+        lines['b2'] /= self.sn / lines['nominal_v'] ** 2
+        return lines.drop('nominal_v', axis=1)
+
+    def get_2_windings_transformers(self) -> _pd.DataFrame:
+        """ Get 2 windings transformers as a ``Pandas`` data frame.
+
+        Returns:
+            a 2 windings transformers data frame
+        """
+        two_windings_transformers = self.get_elements(_pypowsybl.ElementType.TWO_WINDINGS_TRANSFORMER)
+        two_windings_transformers['p1'] /= self.sn
+        two_windings_transformers['q1'] /= self.sn
+        two_windings_transformers['p2'] /= self.sn
+        two_windings_transformers['q2'] /= self.sn
+        two_windings_transformers = _pd.merge(two_windings_transformers, self.get_voltage_levels()['nominal_v'],
+                                              left_on='voltage_level2_id', right_index=True)
+        two_windings_transformers['i1'] /= self.sn * 10 ** 3 / (
+                self.sqrt3 * two_windings_transformers['nominal_v'])
+        two_windings_transformers['i2'] /= self.sn * 10 ** 3 / (
+                self.sqrt3 * two_windings_transformers['nominal_v'])
+        two_windings_transformers['r'] /= two_windings_transformers['nominal_v'] ** 2 / self.sn
+        two_windings_transformers['x'] /= two_windings_transformers['nominal_v'] ** 2 / self.sn
+        two_windings_transformers['g'] /= self.sn / two_windings_transformers['nominal_v'] ** 2
+        two_windings_transformers['b'] /= self.sn / two_windings_transformers['nominal_v'] ** 2
+        two_windings_transformers['rated_u1'] /= two_windings_transformers['nominal_v']
```

Comment:
Need @sylvlecl opinion here: for me ratedU1 is with the nominal voltage on side 1 and ratedU2 is with the nominal voltage en side 2. It is the only way to have a correct rho in the pi model.

Reply:
Indeed ! thank you